### PR TITLE
Use metadata to inject the experiment.html when "experiment": true

### PR DESCRIPTION
**Pull request by juliantoledo**

### CODE REVIEW

**Comment by juliantoledo** [comment]
@sebastianbenz I don't know why, but when I inject experiment.html it does not rendered properly and the variables are empty. Could you take a look?

File: templates/example.html [hunk 1]
```diff
@@ -13,9 +13,12 @@
   <body>
     <div class="body">
       {{> header.html}}
-      {{#sections}}
+      {{#data.sections}}
       <div class="box">
         <div class="column doc {{^inBody}}hide-on-mobile{{/inBody}}">
+          {{#data.experimental}}
+            {{> experiment.html}}
```
Comment (juliantoledo) [inline templates/example.html]: @sebastianbenz here, it where I inject experiment.html and it does not render {{component}}, etc.

**Comment by juliantoledo** [comment]
@sebastianbenz fixed the two questions I asked. PTAL

File: tasks/compile-example.js [hunk 2]
```diff
@@ -60,6 +60,15 @@ module.exports = function(templateRoot, template) {
       const example = ExampleFile.fromPath(file.path);
       const nextExample = example.nextFile();
 
+      // Experimet must be injected in the first section only
```
Comment (sebastianbenz) [inline tasks/compile-example.js]: nit: typo in Experiment

File: tasks/compile-example.js [hunk 3]
```diff
@@ -67,7 +76,8 @@ module.exports = function(templateRoot, template) {
         github: example.githubUrl(),
         subHeading: example.title(),
         exampleStyles: document.styles,
-        sections: document.sections
+        component: example.title(),
```
Comment (sebastianbenz) [inline tasks/compile-example.js]: why are you not using the component from the metadata?

Comment (juliantoledo) [inline tasks/compile-example.js]: I was not sure, I will switch it.

File: templates/example.html [hunk 4]
```diff
@@ -13,9 +13,12 @@
   <body>
     <div class="body">
       {{> header.html}}
-      {{#sections}}
+      {{#data.sections}}
```
Comment (sebastianbenz) [inline templates/example.html]: why do we need the additional 'data' indirection?

Comment (juliantoledo) [inline templates/example.html]: In order to pass:
        "experimental": function() {
          return (idx++ == 0 && document.metadata.experiment);
Experimental needs to be inside the first section.

File: tasks/compile-example.js [hunk 5]
```diff
@@ -60,6 +60,15 @@ module.exports = function(templateRoot, template) {
       const example = ExampleFile.fromPath(file.path);
       const nextExample = example.nextFile();
 
+      // Experimet must be injected in the first section only
+      let idx = 0;
+      const data = {
+        "sections": document.sections,
+        "experimental": function() {
+          return (idx++ == 0 && document.metadata.experiment);
```
Comment (sebastianbenz) [inline tasks/compile-example.js]: Can we rename `experimental` to `inExperiment`? This makes it clear that it is a boolean flag.

Comment (juliantoledo) [inline tasks/compile-example.js]: sure.

File: templates/example.html (hunk 4 above)
Comment (sebastianbenz) [inline templates/example.html]: @juliantoledo understood, but couldn't we just add the function as a field in the args object?

**Comment by jkingyens** [comment]
I have a commit based on this branch that passes this experimental flag to validation task and skips validation for experimental examples. I can wait for this PR to land in master or add the commit here.

**Comment by juliantoledo** [comment]
@jkingyens Sounds good, the metadata is already supported in the document parser. I will finish this PR in a bit.

**Comment by juliantoledo** [comment]
@sebastianbenz fixed your comments and added the metadata instructions to the readme.

File: tasks/compile-example.js [hunk 6]
```diff
@@ -67,9 +66,17 @@ module.exports = function(templateRoot, template) {
...
         subHeading: example.title(),
         exampleStyles: document.styles,
-        sections: document.sections
+        component: document.metadata.component,
+        sections: document.sections,
+        isExperiment: document.metadata.experiment
       };
 
+      if (document.metadata.experiment && (!document.metadata.component ||
+        document.metadata.component.length === 0)) {
```
Comment (sebastianbenz) [inline tasks/compile-example.js]: `if (document.metadata.experiment && !document.metadata.component)` is enough"

```
> false == ''
true
```

Comment (juliantoledo) [inline tasks/compile-example.js]: done.

File: tasks/compile-example.js [hunk 7]
```diff
@@ -67,9 +66,17 @@ module.exports = function(templateRoot, template) {
...
+        throw new Error("Example (" + example.url() +
```
Comment (sebastianbenz) [inline tasks/compile-example.js]: throw a gulp exception: `throw new PluginError(plugin: 'compile-example', message: '...');`

**Comment by sebastianbenz** [comment]
LGTM - good work!

**Pull request merged**